shared/middleware/helpers/assets.rb

```diff
@@ -0,0 +1,68 @@
+#
+# Assets
+#
+class AssetBucket
+
+  ALLOWED_FILE_TYPES = %w(jpg jpeg gif png mp3)
+
+  def initialize
+    params = {region: 'us-east-1'}
+    if CDO.s3_access_key_id && CDO.s3_secret_access_key
+      params[:credentials] = Aws::Credentials.new(CDO.s3_access_key_id, CDO.s3_secret_access_key)
+    end
+    @s3 = Aws::S3::Client.new(params)
```

Comment:
It may be better to cache this client in @@s3, similarly to how @@dynamo_db is cached in DynamoTable. @wjordan do you have an opinion? (sorry Josh I may have suggested the opposite earlier, I'm really not sure which is best)